contractcourt/htlc_outgoing_contest_resolver.go

```diff
@@ -55,38 +55,76 @@ func (h *htlcOutgoingContestResolver) Resolve() (ContractResolver, error) {
 		return nil, nil
 	}
 
+	// handleSpendEvent is a helper closure that processes the spend event
+	// received.
+	handleSpendEvent := func(event *spendResult) (ContractResolver, error) {
+		// Return the error if there's one.
+		if event.err != nil {
+			return nil, event.err
+		}
+
+		// If an event is returned but the spend is nil, then the HTLC
+		// has been spent in a timeout tx. In this case, we'll let the
+		// timeout resolver to handle it.
+		if event.spend == nil {
+			log.Infof("HTLC(%s) has already been spent in "+
+				"timeout tx", h.HtlcPoint())
+			return nil, nil
+		}
```

Comment:
shouldn't we rename the field `spend` to `preimageSpend` in that case?

Reply:
the `spend` in `spendResult` is not necessarily a preimage spent tho, it's just a placeholder for the spending event for that HTLC so it could be either. But yeah it could be confusing here so I renamed the var names in this closure.